lib/elixir/lib/registry.ex

```diff
@@ -209,6 +209,15 @@ defmodule Registry do
   @typedoc "A full match spec used when selecting objects in the registry"
   @type spec :: [{match_pattern, guards, body}]
 
+  @typedoc "Options used for child_spec/1 and start_link/1"
+  @type options :: [
+          keys: keys,
+          name: registry,
+          partitions: pos_integer,
+          listeners: [atom],
+          meta: [{meta_key, meta_value}]
+        ]
```

Comment:
I think that the pattern that we usually use is:

```elixir
@type option :: {:keys, keys} | {:name, registry} | ...

@spec start_link([option]) :: ...
```

Wdyt?

Reply:
Also, I'd probably call this `start_option` instead. :)

Reply:
@whatyouhide updated with your suggestions